File: Day48/DSA/Winner_in_a_election.java
Question:- https://www.geeksforgeeks.org/problems/winner-of-an-election-where-votes-are-represented-as-candidate-names-1587115621/1


// User function Template for Java

class Solution {
    // Function to return the name of candidate that received maximum votes.
    public static String[] winner(String arr[], int n) {
        // add your code
        
        HashMap<String,Integer> mp = new HashMap<>();
        
        for(int i = 0;i<n;i++)
        {
            mp.put(arr[i] , mp.getOrDefault(arr[i] , 0) + 1);
        }
        
        String str = "";
        
        int maxcount = Integer.MIN_VALUE;
        
        for(Map.Entry<String,Integer> ele : mp.entrySet())
        {
            if(ele.getValue() > maxcount)
            {
                maxcount = ele.getValue();
                str = ele.getKey();
            }
            else if(ele.getValue() == maxcount)
            {
                if(str.compareTo(ele.getKey()) > 0)
                {
                    str = ele.getKey();
                }
            }
        }
        
        return new String[]{str , Integer.toString(maxcount)};
    }
}

File: Day48/DSA/Winner_in_a_election.cpp
Question:- https://www.geeksforgeeks.org/problems/winner-of-an-election-where-votes-are-represented-as-candidate-names-1587115621/1

Code:- 

class Solution {
  public:

    // Function to return the name of candidate that received maximum votes.
    vector<string> winner(string arr[], int n) {
        // Your code here
        unordered_map<string , int> mp;
        
        for(int i = 0; i < n; i++)
        {
            mp[arr[i]]++;
        }
        
        int maxcount = INT_MIN;
        
        string str = "";
        
        for(auto i : mp)
        {
            if(i.second > maxcount)
            {
                maxcount = i.second;
                str = i.first;
            }
            else if(i.second == maxcount)
            {
                str = min(i.first,str);
            }
        }
        
        return {str , to_string(maxcount)};
        
        // Return the string containing the name and an integer
        // representing the number of votes the winning candidate got
    }
};